### PR TITLE
fix: catch error if ssm extension is not ready

### PR DESCRIPTION
**Pull request by pudkrong**
Lambda SSM extension might not be ready yet when we request. So, we need to handle properly otherwise it will crash lambda and error loop

### CODE REVIEW

**Comment by coderbyheart** [comment]
This change deals with the lock, nothing with SSM related?

**Comment by pudkrong** [comment]
> This change deals with the lock, nothing with SSM related?

Sorry, I will create a separated PR to fix lock. 

For the SSM cannot read nRF Cloud API Key, the root cause is that reading parameter from outside handler might be failed due to lambda parameter store is not ready yet. Therefore, I reverted the previous changes and commit the new fix.

File: lambda/fetchDeviceShadow.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ const connectionsRepo = connectionsRepository(
 	db,
 	websocketDeviceConnectionsTableName,
 )
-const deviceShadowPromise = (async () => {
+const deviceShadowPromise = once(async () => {
```
Comment (coderbyheart) [inline lambda/fetchDeviceShadow.ts:59]: Unrelated change.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: lambda/fetchDeviceShadow.ts (hunk 1 above)
Comment (coderbyheart) [reply lambda/fetchDeviceShadow.ts:59]: This effectively does the same thing as before. It ensures that deviceShadowPromise only exists once. It will stay in a rejected stated if parameters cannot be fetched.

Comment (coderbyheart) [reply lambda/fetchDeviceShadow.ts:59]: Ah, right. Now I see. It does fix the problem, because the lambda is invoked only in the handler.
